mobileu/tasks.py

```diff
@@ -19,6 +21,15 @@ def send_sms(x, y):
     return x + y
 
 
+def get_teacher_list():
+    teacher_list = TeacherClass.objects.filter(teacher__email__isnull=False)\
+        .distinct('teacher')
+    for rel in teacher_list:
+        if not validate_email(rel.teacher.email):
+            rel.delete()
```

Comment:
I think you might be right. Will change it and make it use exclude to remove the unwanted records.